documentation/proposals/Proposal - Axis Input Devices.md

```diff
@@ -0,0 +1,624 @@
+/// <param name="RawValueBounds">The bounds of the raw value - a value of 'default' indicates it is typical (0, 1), or
+/// in the case that the provided traits are <see cref="AxisTrait.RawValueOnly"/>, this must be default and indicates no known bounds.</param>
+public readonly record struct AxisDescription(
+    int Index,
+    AxisTrait Traits,
+    string Name,
+    Vector2 RawValueBounds = default)
+{
+    public static implicit operator int(AxisDescription description) => description.Index;
+}
+```
+
+### Axis Traits
+These are the primary flags through which an axis's use is defined. Its definition includes several constraints meant to both make defining axes less cumbersome and enforce logically consistent behaviour where possible. This enum is one of the primary places where validation needs to occur at runtime, and some flags place implementation requirements on the implementer. Wherever possible, we should seek to provide default implementations of expected behaviours regarding these constraints.
+
+```csharp
+
+/// <summary>
+/// This is how individual axes are defined
+/// Pay attention to the flags which require specific implementation details related to their RawValues and normalization
+/// </summary>
+[Flags]
+public enum AxisTrait
+{
+    Unknown = 0,
+    Analog = 1 << 0,
+    Binary = 1 << 1,
+
+    // an analog axis that explicitly does not return to center or zero like a joystick or trigger
+    // things like touch-pads, mice, or world-space positions
+    Point = 1 << 8 | Analog | HasRawValue,
+
+    // ---- Physical orientation ----
+    Orientation = 1 << 2 | Analog,
+    Rotation = 1 << 3 | Orientation,
+
+    // this is specified because single axes of euler angle components can be useful by themselves,
+    // whereas anything shorter than 3 or 4 axes is not for quaternion rotation components
+    /// <summary>
+    /// Indicates that this axis is a component of a euler angle (do we want this to be in degrees or radians?)
+    /// If this flag is set, the axis is expected to be normalized between 0-360 degrees, with wrapping applied
+    /// It is required you do NOT set <see cref="AxisDescription.RawValueBounds"/>, as it will be assumed to be (0, 360/2pi).
+    /// </summary>
+    EulerAngleComponent = 1 << 4 | Rotation | HasRawValue,
+
+    /// <summary>
+    /// Acceleration - in meters per second <br/>
+    /// If this is set, it is expected that the axis is NOT provided any raw value constraints,
+    /// as it is not expected to be normalized
+    /// </summary>
+    Acceleration = 1 << 6 | Orientation | RawValueOnly,
+
+    /// <summary>
+    /// Position - raw values are in meters if world-space, in pixels if screen-space, etc <br/>
+    /// Normalized values are required if the bounds are known via the device's API.
+    /// If they are unknown, this must be paired with <see cref="RawValueOnly"/>
+    /// </summary>
+    Position = 1 << 7 | Orientation | Point,

```

Comment:
Again are we assuming that the order of the axes within the group matters so we can interpret these axes as X/Y/Z etc.